pkg/http_endpoint/v1/handlers/http_grafana.go

```
@@ -0,0 +1,337 @@
+package v1handlers
+
+import (
+	"encoding/json"
+	"github.com/pressly/chi"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"visualization-api/pkg/grafanaclient"
+	"visualization-api/pkg/http_endpoint/common"
+	"visualization-api/pkg/logging"
+)
+
+var emailValid = regexp.MustCompile(`^[a-z0-9._%+\-]+@[a-z0-9.\-]+\.[a-z]{2,4}$`)
+
+// V1UsersOrgs implements part of handler interface
+type V1UsersOrgs struct{}
+
+// LoginErrorCheck handles errors for login
+func LoginErrorCheck(err error) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch err.(type) {
+		// InvalidUsernamePassword means, that user provided invalid
+		// username/password. We have to return 401 error then
+		case common.InvalidUsernamePassword:
+			common.WriteErrorToResponse(w, http.StatusUnauthorized,
+				http.StatusText(http.StatusUnauthorized), err.Error())
+			return
+		// If any other error happened -> return 500 error
+		default:
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"Internal server error occured")
+			return
+		}
+	}
+}
+
+// GetUsers get the list of users
+func GetUsers(clients *common.ClientContainer, handler common.HandlerInterface) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		users, err := handler.GetUsers(clients, w)
+		if err != nil {
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"Internal server error occured")
+			return
+		}
+		userlist, _ := json.Marshal(users)
```

serialization error is not handled here